### PR TITLE
Discover bindings made in format spec of format string

### PR DESCRIPTION
**Pull request by jgberry**
Discover bindings made in the format spec of a format string.

Simple example illustrating how a variable can be bound inside a format spec:
```
>>> x
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
NameError: name 'x' is not defined
>>>
>>> f"{3.1415:{(x := 0.3)}}"
'3.14'
>>>
>>> x
0.3
```

### CODE REVIEW

**Review by bwhmather — APPROVED** [review]
Aside from the obvious question of why on earth would you want to do that, this looks spot on.  Looking forward to your `ast.walk` changes.

**Comment by bwhmather** [comment]
I'm sure I must be missing something, but I can't figure out how to trigger the CI workflow and merge this PR from my end.  Are you able to request a build from your end?

**Comment by jgberry** [comment]
Hmm, I did try to trigger the automated checks when I opened the PR but couldn't quite figure it out. I assumed it was something that need to be initiated by you. I don't see anything on my end that would allow me to do so.

**Comment by bwhmather** [comment]
I've tweaked the github actions file to add pull_request as a valid trigger.  I think if you rebase and force push it might work.

Please excuse the fact that this isn't terribly slick.  This is basically my first week of getting contributions and it's taking some figuring out.

**Comment by bwhmather** [comment]
That seems to have done it.  Did you have to click a button to request the run, or was it all on my end?

**Comment by jgberry** [comment]
I believe it was on your end. I pushed an empty commit but was then faced with this message:
Screenshot from 2022-02-25 18-38-50

**Comment by bwhmather** [comment]
Excellent.  I was finding it difficult to believe that this would require maintainers to ask contributors to click a button before merging.  Hopefully the next one should be much smoother.  If you have no objections, I'm going to click the rebase button in the gui and then merge.  Thank you very much, and looking forward to the walk changes.

**Comment by bwhmather** [comment]
Published to pypi as version 0.10.5.

**Pull request merged**
